frontend-react/src/layouts/Markdown/MarkdownLayout.tsx

```diff
@@ -0,0 +1,267 @@
+                (typeof v === "function" && k[0] === k[0].toUpperCase())
+        )
+    );
+}
+
+const uswdsComponents = filterComponents(reactUSWDS);
+const sharedComponents = filterComponents(shared);
+
+export interface MarkdownLayoutProps {
+    children: JSX.Element;
+    frontmatter?: {
+        sidenav?: string;
+        breadcrumbs?: Array<{ label: string; href: string }>;
+        title?: string;
+        subtitle?: string | string[];
+        callToAction?: Array<{ label: string; href: string }>;
+        lastUpdated?: string;
+        toc?: boolean | { depth?: number };
+        backToTop?: boolean;
+    };
+    toc?: TocEntry[];
+    main?: JSX.Element;
+    nav?: JSX.Element;
+    mdx?: React.ComponentProps<typeof MDXProvider>;
+}
+
+/**
+ * Markdown-formatted elements use lowercase variants from this list. If you need manual
+ * control over props, use pascal-cased variants. react-uswds components dynamically added
+ * to list (overridden by project shared components).
+ */
+const MDXComponents = {
+    a: USSmartLink,
+    A: USSmartLink,
+    ...uswdsComponents,
+    ...sharedComponents,
+    Link: USSmartLink,
+    USNavLink,
+    Subtitle: (props: React.PropsWithChildren<{}>) => {
+        return <p>{props.children}</p>;
+    },
+};
+
+/**
+ * Provides a LayoutSidenav component in mdx context that when used will automatically
+ * enable the sidenav of layout.
+ *
+ * FUTURE_TODO: Remove GridContainer once implemented in higher-level component.
+ *
+ * Default markdown layout.
+ * @example
+ * With sidenav:
+ * +------+-----------------+
+ * |      |                 |
+ * |      |                 |
+ * | SIDE |      MAIN       |
+ * |      |                 |
+ * |      |                 |
+ * +------+-----------------+
+ * Without sidenav:
+ * +------------------------+
+ * |                        |
+ * |                        |
+ * |         MAIN           |
+ * |                        |
+ * |                        |
+ * +------------------------+
+ */
+export function MarkdownLayout({
+    children,
+    main,
+    mdx,
+    frontmatter: {
+        title,
+        breadcrumbs,
+        subtitle,
+        callToAction,
+        lastUpdated,
+        toc,
+        backToTop,
+    } = {},
+    toc: tocEntries,
+}: MarkdownLayoutProps) {
+    const helmet = title ? (
+        <Helmet>
+            <title>{title}</title>
+        </Helmet>
+    ) : null;
+    const [sidenavContent, setSidenavContent] =
+        useState<React.ReactNode>(undefined);
+    const [mainContent, setMainContent] = useState<React.ReactNode>(undefined);
+    const ctx = useMemo(() => {
+        return {
+            sidenavContent,
+            setSidenavContent,
+            mainContent,
+            setMainContent,
+        };
+    }, [mainContent, sidenavContent]);
+
+    const tableOfContents =
+        toc && tocEntries ? (
+            <TableOfContents
+                {...(typeof toc === "object" ? toc : {})}
```

Comment:
Both this check and the check below `Array.isArray` seem a little bit like a "code smell" in that they're getting around typing something or we should know something about the data / manipulate before it gets to these functions. But let me know your thoughts!

Reply:
Did some cleanup to separate creating the final expected vars and conditional rendering inside the return.

Reply:
toc is typed in the frontmatter to be either a boolean or an object (to allow for future options [ex: render as a USWDS InPageNav rather than a regular list]) so there's some typing checks that need to happen. This was cleaned up.